Api v5

dchess opened the pull request:
Updates original forked repo to support v5 api with some refactoring.

Commit:
Update ignore with python standards

Commit:
Remove package specific items

Commit:
Move module

Commit:
Add dependencies

Commit:
Add example skeleton

Commit:
Remove README to prepare rewrite

Commit:
Add class based approach skeleton

Commit:
Add example for pulling users

Commit:
Add rule for ignoring raw data

Commit:
Add sqlsorcery dependency

Commit:
Update README with basics

Commit:
Add support for generic-tags

dchess commented:
@mfreemankdc We're building off your original whetstone package to build a version that runs in Docker like the Schoolmint code using the new Whetstone v5 API. Wanted to keep you in the loop as we develop.

Commit:
Add ms sql drivers

Commit:
Add all needed endpoints and tags

zkagin commented on main.py:
```diff
@@ -0,0 +1,50 @@
+from os import path, makedirs
+
+
+def extract(api, endpoint, tag=False):
```
Can you rename the function to something that makes it more clear what this is doing?

dchess replied:
Agreed. This has been deprecated in #2.

zkagin commented on main.py:
```diff
@@ -0,0 +1,50 @@
+
+def extract(api, endpoint, tag=False):
+    filename = f"data/{endpoint}.json"
+    if path.exists(filename):
```
This is probably easier to read as `if not path.exists(filename):` so you don't need an `else`.

dchess replied:
Makes sense.

zkagin commented on main.py:
```diff
@@ -0,0 +1,50 @@
+
+def main():
+    api = Whetstone()
+    endpoints = [
+        "assignments",
+        "informals",
+        "measurements",
+        "meetings",
+        "observations",
+        "rubrics",
+        "schools",
+        "users",
+    ]
+
+    for endpoint in endpoints:
+        extract(api, endpoint)
+
+    tags = [
+        "grades",
+        "courses",
+        "measurementgroups",
+        "goaltypes",
+        "meetingtypes",
+        "observationtypes",
+        "assignmenttypes",
+        "assignmentpresets",
+    ]
+
+    for tag in tags:
+        extract(api, tag, tag=True)
```
I think this might read a little simpler if it was iterating through a list of tuples where the second component was whether or not it was a tag. What do you think?

dchess replied:
@zkagin I've refactored this approach in #2.

zkagin commented on whetstone.py:
```diff
@@ -0,0 +1,53 @@
+    def __init__(self, qa=False):
+        self.client_id = os.getenv("CLIENT_ID")
+        self.client_secret = os.getenv("CLIENT_SECRET")
+        if qa:
+            subdomain = "api-qa"
+        else:
+            subdomain = "api"
```
This is a perfect spot for a ternary function if you want to use it.
`subdomain = "api-qa" if qa else "api"`

dchess replied:
I like this. I will change it in #2.

zkagin reviewed (changes requested):
Generally looks great! I had a few minor comments that might make things a little more readable, but otherwise it all seems pretty clear.

One broader request: Can you add docstrings to all the major functions?

zkagin reviewed (approved):
Looks good, given that a chunk of this gets rewritten in the next PR.

Going forward I think it'd be helpful to try and break PRs up a little more so that we're not left with code we don't like between PRs, and to keep the changes a little smaller and easier to review. The first few are a pretty big rewrite, so it's hard to break that up right now anyways.

dchess commented:
> Looks good, given that a chunk of this gets rewritten in the next PR.
> 
> Going forward I think it'd be helpful to try and break PRs up a little more so that we're not left with code we don't like between PRs, and to keep the changes a little smaller and easier to review. The first few are a pretty big rewrite, so it's hard to break that up right now anyways.

Agreed. This was opened quite a while ago and was really meant to get the overall setup and test out the api. I can close this PR since #2 encapsulates it.

zkagin commented:
@dchess Is the plan to merge any relevant changes into #2, or to land this as is? We should update #2 if the former is the plan.

Commit:
Endpoints (#2)

Depends on #1

Adds logic for writing endpoints to database parsed into a relation model.
